### PR TITLE
Proposed fix 1416 Drag-and-drop of automatables to Automation Editor

### LINKED ISSUE #1416: Drag-and-drop of automatables to Automation Editor
I propose the automation CTRL+Drag process allows the same behavior if a control is dragged to a editor as it does if it is dragged to a block.

screen shot 2014-12-06 at 2 46 45 pm

### PR DESCRIPTION
**Pull request by curlymorphic**
Proposed fix #1416 Drag-and-drop of automatables to Automation Editor

### CODE REVIEW

File: src/gui/editors/AutomationEditor.cpp [hunk 1]
```diff
@@ -2220,6 +2222,7 @@ AutomationEditorWindow::AutomationEditorWindow() :
 	setFocusPolicy( Qt::StrongFocus );
 	setFocus();
 	setWindowIcon( embed::getIconPixmap( "automation" ) );
+	setAcceptDrops( true );
```
Comment (tresf) [inline src/gui/editors/AutomationEditor.cpp]: Nitpicking here... it appears we generally link the toolbar as well... 

https://github.com/LMMS/lmms/blob/master/src/gui/editors/BBEditor.cpp#L56

File: src/gui/editors/AutomationEditor.cpp [hunk 2]
```diff
@@ -2282,6 +2285,35 @@ const AutomationPattern* AutomationEditorWindow::currentPattern()
...
+				Engine::projectJournal()->
+					journallingObject( val.toInt() ) );
+		if( mod != NULL )
+		{
+			if( m_editor->m_pattern->firstObject() )
+			{
+				m_editor->m_pattern->objectDestroyed( m_editor->m_pattern->firstObject()->id() );
+			}
+			m_editor->m_pattern->clear();
+			 m_editor->m_pattern->addObject( mod );
```
Comment (tresf) [inline src/gui/editors/AutomationEditor.cpp]: ` :) extra space  ---------------------^`

File: src/gui/editors/AutomationEditor.cpp [hunk 3]
```diff
@@ -2282,6 +2285,35 @@ const AutomationPattern* AutomationEditorWindow::currentPattern()
...
+		AutomatableModel * mod = dynamic_cast<AutomatableModel *>(
+				Engine::projectJournal()->
+					journallingObject( val.toInt() ) );
+		if( mod != NULL )
+		{
+			if( m_editor->m_pattern->firstObject() )
+			{
+				m_editor->m_pattern->objectDestroyed( m_editor->m_pattern->firstObject()->id() );
+			}
+			m_editor->m_pattern->clear();
```
Comment (tresf) [inline src/gui/editors/AutomationEditor.cpp]: @diizy do we normally clear the pattern when we drag in a new item?  Currently we can link one automation event to multiple similar controls, right?

Comment (curlymorphic) [inline src/gui/editors/AutomationEditor.cpp]: > Currently we can link one automation event to multiple similar controls, right?

Now i understand what the code is doing, changing this now

**Comment by curlymorphic** [comment]
Removed to code that only linked one controller, linked to toolbar, and minor reformatting

**Comment by tresf** [comment]
This looks great.  One possible wish...

The display doesn't seem to update properly when dragged directly in.  The window needs to be closed and reopened in order to refresh properly.

image

**should look like**

image

**Comment by curlymorphic** [comment]
> The display doesn't seem to update properly when dragged directly in.

Sorted :)

**Comment by badosu** [comment]
Nice one @curlymorphic! As always :-)

**Comment by tresf** [comment]
Clean and simple.  Merging.  Thanks!

**Pull request merged**